examples/matlab/time_freezing_hystheresis/time_optimal_control_car/main_car_hysteresis_ocp.m

```diff
@@ -46,15 +46,19 @@
 settings.local_speed_of_time_variable = 1;
 % solver settings
 settings.opts_ipopt.ipopt.tol = 1e-8;
-settings.comp_tol = 1e-8;
+settings.comp_tol = 1e-5;
 settings.cross_comp_mode = 8;
 settings.homotopy_update_rule = 'superlinear';
 
+settings.initial_lambda = 0.0;
+settings.initial_theta = 0.0;
+settings.initial_mu = 0.0;
+
 %% Model Settings
 model.fuel_cost_on = 0;
 model.N_finite_elements = 3;
 model.N_stages = 10;
-model.T = 5;
+model.T = 1;
```

Comment:
This one is not unnecessary. One can still provide a `T` in a time optimal problem as well.
This is the length of the "numerical time" and it can be different than 1. 
Mathematically, there optimal time is `T_opt = T*s_sot` . There is some ambiguity of course. 
You can always set `T = 1` (or any number you like) and let `s_sot` do the rest of the job.
However, in practice putting some higher/different `model.T`  can improve the convergence, as you iterate less over `s_sot` and the problem is quite nonlinear in these directions.